test/unit/lodash.spec.js

```diff
@@ -0,0 +1,690 @@
+                    if (typeof arr[i] !== 'string') {
+                        expect(typeof arr[i]).to.eql('string');
+                    }
+                    if (arr[i].length === 3) {
+                        string0++;
+                    } else if (arr[i].length === 4) {
+                        string1++;
+                    } else if (arr[i].length === 5) {
+                        string2++;
+                    }
+                }
+
+                expect(arr.length).to.eql(6);
+                expect(string0).to.eql(2);
+                expect(string1).to.eql(3);
+                expect(string2).to.eql(1);
+            });
+            it('should count length of given strings', function () {
+                expect(_.countBy.apply(_, datasets.countBy2())).to.eql({3: 2, 4: 3, 5: 1});
+            });
+        });
+    });
+
+    describe('every', function () {
+        describe('every function', function () {
+            var params = datasets.every();
+
+            it('should match types of passing elements', function () {
+                var element1 = params[0] instanceof Array;
+                var element2 = typeof params[1];
+
+                expect(element1).to.eql(true);
+                expect(element2).to.eql('string');
+                expect(params.length).to.eql(2);
+            });
+            it('should contain array of objects', function () {
+                var array = params[0];
+                for (var i = 0; i < array.length; i++) {
+                    expect(Object.keys(array[i]).length).to.eql(2);
+                }
+
+                expect(array.length).to.eql(3);
+                array.forEach(function (n) {
+                    expect(_.has(n, 'ALIVE')).to.eql(true);
+                });
+            });
+            it('should return true if all elements pass the predicate "ALIVE" check', function () {
+                expect(_.every.apply(_, datasets.every())).to.eql(true);
+            });
+        });
+        describe('every2 function', function () {
+            var params = datasets.every2();
+            var array = params[0];
+            var arrLenght = array.length;
+
+            it('should match types of passing elements', function () {
+                var element1 = params[0] instanceof Array;
+                var element2 = params[1] instanceof Function;
+
+                expect(element1).to.eql(true);
+                expect(element2).to.eql(true);
+                expect(params.length).to.eql(2);
+            });
+            it('should contain array of different type elements which are Booleanly true', function () {
+                var types = _.map(array, function (i) {
+                    array = typeof i;
+
+                    return array;
+                });
+                var result = _.uniq(types);
+
+                expect(result.length).to.eql(arrLenght);
+            });
+
+            it('should return true if all elements are Booleanly true', function () {
```

Comment:
That is quite good test case but iteratee function hasn't been tested at all and I can pass whatever I want.